src/main/scala/Dispatcher.scala

```diff
@@ -0,0 +1,41 @@
+import scala.io.Source
+
+import telegram.API._
+import tinkoff.API._
+
+/**
+  * Main polling loop.
+  */
+object Dispatcher extends App {
+
+  def getFormattedRates(rates: List[Rate]): String = {
+    val textRatesList = rates.map(x => "%3s: %7.3f  %7.3f".format(x.fromCurrency.name, x.buy, x.sell.get))
+    "<pre>" +
+      "       Buy      Sell\n" +
+      "      ------   ------\n" +
+      textRatesList.mkString("\n") + "</pre>"
+  }
+
+  setToken(Source.fromFile("TelegramBotToken").getLines.mkString)
+
+  while (true) {
+    val updatesList = getUpdates
+
+    for (update <- updatesList; if update.message.isDefined; message = update.message.get) {
+        if (message.entities.isDefined)
+          for (entity <- message.entities.get if entity.`type` == "bot_command") {
+            // (jusual): message.text is defined because text contains name of bot_command
+            val command = message.text.get.substring(entity.offset, entity.offset + entity.length)
```

Comment:
Global iteration is over entities. So you should write like this: message.entities.map(x =>
if x.type == "bot_command "; x.foreach(message.text.map(_.substring(x.offset , x.offset + x.length))))).filter(_ == "/rates").foreach(...).
So, it is very long unreadable line. 
Map and foreach only for command doesn't make sense.